src/cgroup.rs

```diff
@@ -346,7 +346,16 @@ impl Cgroup {
             let subsystems = self.subsystems();
             if !subsystems.is_empty() {
                 let c = subsystems[0].to_controller();
-                c.add_task(&tid)
+                let cgroup_type = self.get_cgroup_type()?;
```

Comment:
Just wondering if could be better if we put the determine inside the inner add_task.